src/Service.Tests/Configuration/ConfigurationTests.cs

```diff
@@ -1388,6 +1414,242 @@ public async Task TestSchemaIntrospectionQuery(bool enableIntrospection, bool ex
+        /// <param name="endpoint">The endpoint route</param>
+        /// <param name="hostModeType">The mode in which the service is executing.</param>
+        /// <param name="expectedStatusCode">Expected Status Code.</param>
+        /// <param name="expectedContent">The expected phrase in the response body.</param>
+        [DataTestMethod]
+        [TestCategory(TestCategory.MSSQL)]
+        [DataRow("/swagger", HostModeType.Development, HttpStatusCode.OK, "", DisplayName = "SwaggerUI enabled in development mode.")]
+        [DataRow("/swagger", HostModeType.Production, HttpStatusCode.BadRequest, "", DisplayName = "SwaggerUI disabled in production mode.")]
+        public async Task OpenApi_InteractiveSwaggerUI(
+            string endpoint,
+            HostModeType hostModeType,
+            HttpStatusCode expectedStatusCode,
+            string expectedContent = "")
+        {
+            Dictionary<GlobalSettingsType, object> settings = new()
+            {
+                { GlobalSettingsType.Host, JsonSerializer.SerializeToElement(new HostGlobalSettings(){ Mode = hostModeType}) },
+                { GlobalSettingsType.GraphQL, JsonSerializer.SerializeToElement(new GraphQLGlobalSettings(){ Enabled = true }) },
+                { GlobalSettingsType.Rest, JsonSerializer.SerializeToElement(new RestGlobalSettings(){ Enabled = true }) }
+            };
+
+            DataSource dataSource = new(DatabaseType.mssql)
+            {
+                ConnectionString = GetConnectionStringFromEnvironmentConfig(environment: TestCategory.MSSQL)
+            };
+
+            RuntimeConfig configuration = InitMinimalRuntimeConfig(globalSettings: settings, dataSource: dataSource);
+            const string CUSTOM_CONFIG = "custom-config.json";
+            File.WriteAllText(
+                CUSTOM_CONFIG,
+                JsonSerializer.Serialize(configuration, RuntimeConfig.SerializerOptions));
+
+            string[] args = new[]
+            {
+                    $"--ConfigFileName={CUSTOM_CONFIG}"
+            };
+
+            TestServer server = new(Program.CreateWebHostBuilder(args));
+
+            HttpClient client = server.CreateClient();
+            HttpRequestMessage request = new(HttpMethod.Get, endpoint);
+
+            // Adding the following headers simulates an interactive browser request.
+            request.Headers.Add("user-agent", "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/105.0.0.0 Safari/537.36");
+            request.Headers.Add("accept", "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9");
+
+            HttpResponseMessage response = await client.SendAsync(request);
+            Assert.AreEqual(expectedStatusCode, response.StatusCode);
+            string actualBody = await response.Content.ReadAsStringAsync();
+            Assert.IsTrue(actualBody.Contains(expectedContent));
```

Comment:
Is expectedContent empty even when swagger Ui is enabled in development mode?

Reply:
no. Updated to accommodate swagger endpoint redirect and html payload. will also double as a test on whether swagger endpoint is honoring the custom rest config.